### PR TITLE
Documents `hemispheres` behaviour for odd size templates

### PR DESCRIPTION
**Pull request by IgorTatarnikov**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/brainglobe/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
Currently it's possible that a symmetric template might have a shape that's odd in the `frontal` axis. The behaviour was undocumented.

**What does this PR do?**
Documents the behaviour for this edge case and adds tests.

## How has this PR been tested?
New tests added that check the behaviour with an even and odd shaped template.

## Is this a breaking change?
No

## Does this PR require an update to the documentation?
No

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: brainglobe_atlasapi/core.py [hunk 1]
```diff
@@ -124,6 +124,14 @@ def annotation(self):
 
     @property
     def hemispheres(self):
+        """
+        Returns a stack with the hemisphere information. 1 - left, 2 - right.
+
+        If a symmetric reference is used, the hemisphere information is
+        generated by splitting the reference in half along the frontal axis.
+        If the reference has an odd number of voxels along the frontal axis,
+        the middle plane is assigned to the left hemisphere.
```
Comment (VascoSch92) [inline brainglobe_atlasapi/core.py:133]:
```suggestion
        Returns a stack with the hemisphere information: 1 for the left hemisphere and 2 for the right.

        If a symmetric reference is used, the hemisphere information is
        determined by splitting the reference in half along the frontal axis.
        If the reference has an odd number of voxels along the frontal axis,
        the middle plane is assigned to the left hemisphere.
```

Comment (alessandrofelder) [reply brainglobe_atlasapi/core.py:133]: Thanks @VascoSch92 - I don't think this change adds much. The hemispheres are generated on-the-fly as the function is called, so I think using the word "generated" is more precise. Again, please get in touch through zulip so we can coordinate efforts if you'd like to contribute :)

**Review by alessandrofelder — APPROVED** [review]
